src/jit/codegencommon.cpp

```diff
@@ -10279,12 +10326,39 @@ void CodeGen::genFuncletProlog(BasicBlock* block)
 
     ScopedSetVariable<bool> _setGeneratingProlog(&compiler->compGeneratingProlog, true);
 
+    gcInfo.gcResetForBB();
+
     compiler->unwindBegProlog();
 
-    // TODO Save callee-saved registers
+    inst_RV(INS_push, REG_FPBASE, TYP_REF);
+    compiler->unwindPush(REG_FPBASE);
 
     // This is the end of the OS-reported prolog for purposes of unwinding
     compiler->unwindEndProlog();
+
+    // Use PSPSym value as a flag for restoring EBP
+    if (compiler->lvaPSPSym != BAD_VAR_NUM)
+    {
+        if (block->bbCatchTyp != BBCT_FINALLY && block->bbCatchTyp != BBCT_FAULT)
+        {
+            regTracker.rsTrackRegTrash(REG_FPBASE);
+
+            // TODO 16 byte stack alignment
+            // TODO Fix, if needed, for nested try-catch
+
+            // mov ebp, [esp + 8]  ; get ebp from caller argument
+            getEmitter()->emitIns_R_AR(INS_mov, EA_PTRSIZE, REG_FPBASE, REG_SPBASE, 8);
+            // lea ebp, [ebp - 8]  ; load ebp from main function
+            getEmitter()->emitIns_R_AR(INS_lea, EA_PTRSIZE, REG_FPBASE, REG_FPBASE, -8);
+            // mov eax, [esp + 12] ; exception object to eax: funclet needs eax having the exception object
+            getEmitter()->emitIns_R_AR(INS_mov, EA_PTRSIZE, REG_EAX, REG_SPBASE, 12);
+
```

Comment:
It would be better to use ``REG_EXCEPTION_OBJECT`` instead of ``REG_EAX`` (although they are same).